test/Upgrades.t.sol

```diff
@@ -0,0 +1,679 @@
+        managerChain1.setOutboundLimit(type(uint64).max);
+        managerChain1.setInboundLimit(type(uint64).max, chainId2);
+
+        // Chain 2 setup
+        vm.chainId(chainId2);
+        DummyToken t2 = new DummyTokenMintAndBurn();
+        ManagerStandalone implementationChain2 =
+            new ManagerStandalone(address(t2), Manager.Mode.BURNING, chainId2, 1 days);
+            
+        managerChain2 = ManagerStandalone(address(new ERC1967Proxy(address(implementationChain2), "")));
+        managerChain2.initialize();
+
+        WormholeEndpointStandalone wormholeEndpointChain2Implementation = new WormholeEndpointStandalone(
+            address(managerChain2), 
+            address(wormhole),
+            address(relayer)
+        );
+        wormholeEndpointChain2 = WormholeEndpointStandalone(address(new ERC1967Proxy(address(wormholeEndpointChain2Implementation), "")));
+        wormholeEndpointChain2.initialize();
+
+        managerChain2.setEndpoint(address(wormholeEndpointChain2));
+        managerChain2.setOutboundLimit(type(uint64).max);
+        managerChain2.setInboundLimit(type(uint64).max, chainId1);
+
+        // Register sibling contracts for the manager and endpoint. Endpoints and manager each have the concept of siblings here.
+        managerChain1.setSibling(chainId2, bytes32(uint256(uint160(address(managerChain2)))));
+        managerChain2.setSibling(chainId1, bytes32(uint256(uint160(address(managerChain1)))));
+
+        wormholeEndpointChain1.setWormholeSibling(chainId2, bytes32(uint256(uint160((address(wormholeEndpointChain2))))));
+        wormholeEndpointChain2.setWormholeSibling(chainId1, bytes32(uint256(uint160(address(wormholeEndpointChain1)))));
+
+        managerChain1.setThreshold(1);
+        managerChain2.setThreshold(1);
+        vm.chainId(chainId1);
+    }
+
+    function testBasicUpgradeManager() public{
+
+        // Basic call to upgrade with the same contact as ewll
+        ManagerStandalone newImplementation = new ManagerStandalone(address(managerChain1.token()), Manager.Mode.LOCKING, chainId1, 1 days);
+        managerChain1.upgrade(address(newImplementation)); 
+
+        basicFunctionality();
+    }
+
+    //Upgradability stuff for endpoints is real borked because of some missing implementation. Test this later once fixed.
+    function testBasicUpgradeEndpoint() public{
+
+        // Basic call to upgrade with the same contact as well
+        WormholeEndpointStandalone wormholeEndpointChain1Implementation = new WormholeEndpointStandalone(
+            address(managerChain1), 
+            address(wormhole),
+            address(relayer)
+        ); 
+        managerChain1.upgradeEndpoint(address(wormholeEndpointChain1), address(wormholeEndpointChain1Implementation)); 
+
+        basicFunctionality();
+    }
+
+
+    // Confirm that we can handle multiple upgrades as a manager
+    function testDoubleUpgradeManager() public {
```

Comment:
Nit-pick, but can all the test be named like `test_camelCase`

Reply:
The linter lied to me. Can do!